integrations/spiral/src/Bootloader/SearchBootloader.php

```diff
@@ -82,144 +95,82 @@ public function boot(ConsoleBootloader $console, Container $container): void
-        }
+        return $engines['default'] ?? \reset($engines);
+    }
 
-        if (\class_exists(TypesenseAdapterFactory::class)) {
-            $container->bindSingleton(TypesenseAdapterFactory::class, fn (Container $container) => new TypesenseAdapterFactory($container));
+    private function createAdapterFactory(Container $container): AdapterFactory
+    {
+        $adapterServices = []; // TODO tagged services would make this extensible
 
-            $adapterServices[TypesenseAdapterFactory::getName()] = TypesenseAdapterFactory::class;
+        foreach (self::ADAPTER_FACTORIES as $adapterClass) {
+            if (!\class_exists($adapterClass)) {
+                continue;
+            }
+            
+            $container->bindSingleton($adapterClass, $adapterClass);
+            $adapterServices[$adapterClass::getName()] = $adapterClass;
         }
 
         // ...
 
+        $prefix = 'schranz_search.adapter.';
+
         if (\class_exists(ReadWriteAdapterFactory::class)) {
-            $container->bindSingleton(ReadWriteAdapterFactory::class, fn (Container $container) => new ReadWriteAdapterFactory(
-                $container,
-                'schranz_search.adapter.',
-            ));
+            $container->bindSingleton(
+                ReadWriteAdapterFactory::class,
+                static fn (Container $container) => new ReadWriteAdapterFactory($container, $prefix),
+            );
 
             $adapterServices[ReadWriteAdapterFactory::getName()] = ReadWriteAdapterFactory::class;
         }
 
         if (\class_exists(MultiAdapterFactory::class)) {
-            $container->bindSingleton(MultiAdapterFactory::class, fn (Container $container) => new MultiAdapterFactory(
-                $container,
-                'schranz_search.adapter.',
-            ));
+            $container->bindSingleton(
+                MultiAdapterFactory::class,
+                static fn (Container $container) => new MultiAdapterFactory($container, $prefix),
+            );
 
             $adapterServices[MultiAdapterFactory::getName()] = MultiAdapterFactory::class;
         }
 
         // ...
 
-        $container->bindSingleton(AdapterFactory::class, function (Container $container) use ($adapterServices) {
-            $factories = [];
-            foreach ($adapterServices as $name => $adapterServiceId) {
-                /** @var AdapterFactoryInterface $adapterFactory */
-                $adapterFactory = $container->get($adapterServiceId);
+        $factories = [];
+        foreach ($adapterServices as $name => $adapterServiceId) {
+            /** @var AdapterFactoryInterface $adapterFactory */
+            $adapterFactory = $container->get($adapterServiceId);
 
-                $factories[$name] = $adapterFactory;
-            }
+            $factories[$name] = $adapterFactory;
+        }
 
-            return new AdapterFactory($factories);
-        });
```

Comment:
Looks unexpected for me to throw this aways as this avoids that a ervice can be decorated over the container and from my point of view that is the whole part why using a dependency injection container to allow manipulation of the services.